### PR TITLE
Render saloon

### PR DESCRIPTION
**Pull request by MatsJoh50**
**Feature to render the saloon based on blueprint from DB:**

This feature involves rendering the saloon based on a blueprint retrieved from the database. It includes a "multi-hover" functionality that adds and removes opacity-50 to highlight the chosen seats while hovering over the leftmost seat. Additionally, it's designed to be quickly adaptable to future components, such as one to specify the number of seats to book. 

edit: new push made some changes as follow: 

- noGo seats are displayed as dark red
- chosen and clicked seats are displayed as white
- footer got "relative" instead of "fixed" also added "mt-4"
- moved some files according to group meeting.

clickedSeats_and_noGo

(Seats 14, 15, 16, and 17 are highlighted in the screenshot as they are hovered over.)

**Close-together Feature  "Seats to Book":**

This feature allows users to set the seats to book by clicking on them. It grabs the hovered seats and pushes them to a useState-array.

**Features to Continue Adding:**

- **Limit of Row:** Implement a limit for the number of rows. This should be straightforward to continue with, as the prerequisites are already in place.
- **Check if Seats are Taken onClick:** Check whether seats are already booked when clicked. This will require the booking schema and database to exist but shouldn't take much work after that.
- **Add Key to Special Seats to Identify by Booking:** Implement a key to identify special seats booked. There's an idea for this, but it hasn't been executed yet.

Feel free to ask for any further clarification or assistance with the language!

### CODE REVIEW

File: kinogoteborg/src/app/booking/page.jsx
Comment (AnnaLenaO) [inline kinogoteborg/src/app/booking/page.jsx:1]: Maybe this file can be moved to a new folder [movieID] in folder booking to match the link between a movie and booking in movie/Movie.jsx row 46.

File: kinogoteborg/src/app/components/booking/clickOutside.jsx
Comment (AnnaLenaO) [inline kinogoteborg/src/app/components/booking/clickOutside.jsx:1]: This looks like a nice feature. I can not find it exported and then imported somevere else. Maybe I missed it or it is not implemented yet.

**Review by AnnaLenaO — COMMENTED** [review]
Great work with all the logic for render the saloon. 
Very nice with a mouseOver hover effect!
I left some comments about what we discussed as possible alternatives for file structure to connect the booking page with a movie.

File: kinogoteborg/src/app/components/booking/clickOutside.jsx
Comment (MatsJoh50) [reply kinogoteborg/src/app/components/booking/clickOutside.jsx:1]: this is my fault of bad seperation of PBI's, so this feature isnt working nor is it implemented anywhere atm. (shoulnt be anyway)

File: kinogoteborg/src/app/booking/screeningSchema.js
[the remark this answers is not in the document]
Comment (MatsJoh50) [reply kinogoteborg/src/app/booking/screeningSchema.js:1]: done and done

**Review by Kookiie88 — APPROVED** [review]
I checked out your branch and everything seems to look correct based on your description of the changes!

**Comment by Weeezzt** [comment]
As you can se below, it looks like you cant book the seats when you hovering on the seat next to the handicaps, but if you hover on the next seat on the row, suddenly you can book them.. Everything else looks good, proud of you man!

Screenshot 2024-05-15 at 13 28 40


Screenshot 2024-05-15 at 13 28 45

**Pull request merged**